### PR TITLE
[DEV] allow configuration of context logging level

### PR DESCRIPTION
**Pull request by evanmags**
- adds configurable context entrance and exit logging.
- updates docs with new environment variable
- adds release notes to changelog

### CODE REVIEW

File: woodchipper/context.py [hunk 1]
```diff
@@ -138,12 +139,22 @@ def _convert_to_loggable_value(value: Any) -> LoggableValue:
             )
 
 
+def _convert_to_valid_log_level(log_level: Union[str, int, Missing]) -> int:
```
Comment (evanmags) [inline woodchipper/context.py:142]: I don't love this as it feels too hand-hold-y. But failing fast is not really an option as the value can be updated at any point. This guaranties that any value passed to a logging context as a log level is made valid, so we do not throw when we get to the logging call.

File: docs/advanced.md [hunk 2]
```diff
@@ -41,6 +41,31 @@ with LoggingContext(user=request.user.id, _prefix="demo"):
...
+
+You can set the default log level for all context logs by setting the environment variable `WOODCHIPPER_CONTEXT_LOG_LEVEL` to
+the desired log level name. For example, if you set `WOODCHIPPER_CONTEXT_LOG_LEVEL` to `"DEBUG"` all logs created when entering or
+exiting a LoggingContext would be debug by default.
+
+You can also set the log level on a `LoggingContext`, overriding the default log level, with the `_log_level` kwarg. For
+example:
+
+```python
+with LoggingContext(user=request.user.id, _log_level="ERROR"):
```
Comment (erinorourke17) [inline docs/advanced.md:56]: Maybe this is a python convention I'm not familiar with, but why use an underscore in front of the _log_level parameter? It's not a private attribute of a class, right?

**Review by erinorourke17 — COMMENTED** [review]
I think this is a reasonable change! Happy to approve when it's non-draft. Aside from my underscore question, I realized the last PR to this repo went in without a corresponding ticket. Do you have one to track the changes

File: docs/advanced.md (hunk 2 above)
Comment (j00bar) [reply docs/advanced.md:56]: To indicate that it's not a context variable that should be added to the logging context but instead a configuration directive for `LoggingContext`.

File: woodchipper/context.py (hunk 1 above)
Comment (j00bar) [reply woodchipper/context.py:144]: I like it.

File: docs/advanced.md (hunk 2 above)
Comment (erinorourke17) [reply docs/advanced.md:56]: Ahh got it, thank you!

**Review by erinorourke17 — APPROVED** [review]

**Pull request merged**